tests/test_rabfe_topology.py

```diff
@@ -162,6 +162,46 @@ def test_dual_topology_standard_decoupling():
     np.testing.assert_array_equal(combined_lambda_offset_idxs[A:], np.ones(B))
 
 
+def test_dual_topology_standard_decoupling_charged():
+
+    # special test case for charged molecules, we expect the charges to be rescaled
+    # based on each individual molecule's charge, as opposed to based on the sum
+    # of the charges.
+
+    ff_handlers = deserialize_handlers(open("ff/params/smirnoff_1_1_0_sc.py").read())
+    ff = Forcefield(ff_handlers)
+
+    mol_a = Chem.AddHs(Chem.MolFromSmiles("C1CC1[O-]"))
+    mol_b = Chem.AddHs(Chem.MolFromSmiles("C1[O+]CCCCC1"))
+    mol_c = Chem.CombineMols(mol_a, mol_b)
+    mol_top = topology.DualTopologyStandardDecoupling(mol_a, mol_b, ff)
+
+    qlj_params, nonbonded_potential = mol_top.parameterize_nonbonded(ff.q_handle.params, ff.lj_handle.params)
+
+    assert isinstance(nonbonded_potential, potentials.NonbondedInterpolated)
+
+    expected_qlj = topology.standard_qlj_typer(mol_c)
```

Comment:
Might be good to add in an assertion that this does not match the output of parameterize, right now it is unclear that this is doing the wrong thing

Reply:
I'll just build this by hand.